### PR TITLE
[CI; Tune] Split Tune tests and examples

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Why are these changes needed?

Split Tune tests into multiple chunks.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -615,17 +615,38 @@
...
-- label: ":octopus: Tune tests and examples {1/2; no RLlib}"
+- label: ":octopus: Tune tests {A-P; no RLlib}"
+  conditions: ["RAY_CI_TUNE_AFFECTED"]
+  commands:
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/travis/upload_build_info.sh; fi }; trap cleanup EXIT
+    - TUNE_TESTING=1 PYTHON=3.7 ./ci/travis/install-dependencies.sh
+    # Because Python version changed, we need to re-install Ray here
+    - rm -rf ./python/ray/thirdparty_files; rm -rf ./python/ray/pickle5_files; ./ci/travis/ci.sh build
+    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only
+      --test_tag_filters=tests_dir_A,tests_dir_B,tests_dir_C,tests_dir_D,tests_dir_E,tests_dir_F,tests_dir_G,tests_dir_H,tests_dir_I,tests_dir_J,tests_dir_K,tests_dir_L,tests_dir_M,tests_dir_N,tests_dir_O,tests_dir_P,-example,-flaky,-py37,-soft_imports,-gpu_only,-rllib
```
Comment (richardliaw) [inline .buildkite/pipeline.yml:626]: can you instead to `test_dir_A-P` and `test_dir_Q-Z`?

Comment (xwjiang2010) [reply .buildkite/pipeline.yml:626]: I like the more granular tags, so that we can easily shift load around or split more when we add more tests.

Comment (sven1977) [reply .buildkite/pipeline.yml:637]: I'm with @xwjiang2010. Will make it easier later to keep this organized when as tests come aboard.

**Review by sven1977 — APPROVED** [review]
Looks great. Let's merge it. Thanks @xwjiang2010 !

**Pull request merged**